WcaOnRails/db/migrate/20231122153755_create_wfc_dues_redirects.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+class CreateWfcDuesRedirects < ActiveRecord::Migration[7.0]
+  def change
+    create_table :wfc_dues_redirects do |t|
+      t.string :redirect_source_id, polymorphic: true, null: false
+      t.string :redirect_source_type, polymorphic: true, null: false
```

Comment:
If you manually hack this, the `polymorphic: true` option is not necessary. It is only specific to `t.references`, not to `t.string` or `t.integer`.

Have you tried `t.references :redirect_source, polymorphic: true, type: :string`?